### PR TITLE
Implementation of unit test

### PR DESCRIPTION
**Pull request by ezequielsetzesglobant**
The required unit tests were implemented

image

image

### CODE REVIEW

File: app/src/main/java/com/example/parkingsystem/mvp/model/ReservationModel.java [hunk 1]
```diff
@@ -52,7 +55,7 @@ public void saveReservation(String securityCode, String place) {
     }
 
     private boolean isOverlap(Reservation reservation) {
-        List<Reservation> reservationList = reservationInformationDB.getReservationsDB(reservation.getPlace());
+         List<Reservation> reservationList = reservationInformationDB.getReservationsDB(reservation.getPlace());
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/model/ReservationModel.java]: is this space necessary?

File: app/src/test/java/com/example/parkingsystem/mvp/presenter/ReservationPresenterTest.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    public static final int HOUR_START = 3;
+    public static final int MINUTE_START = 17;
+
+    public static final int YEAR_FINISH = 2022;
+    public static final int MONTH_FINISH = 9;
+    public static final int DAY_OF_MONTH_FINISH = 10;
+    public static final int HOUR_FINISH = 6;
+    public static final int MINUTE_FINISH = 25;
+
+    @Mock
```
Comment (SartiMau) [inline app/src/test/java/com/example/parkingsystem/mvp/presenter/ReservationPresenterTest.java:42]: please use only one type of mock.
Use the annotation `@Mock` to the  `= mock(class)`

File: app/src/test/java/com/example/parkingsystem/mvp/presenter/ReservationPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    @Test
+    public void createDateTest() {
+        presenter.createDate(listenerDate, true);
+        verify(view).showDatePicker(listenerDate);
+    }
+
+    private void saveReservationDateTime() {
+        model.setStartDateAndTime(true);
```
Comment (SartiMau) [inline app/src/test/java/com/example/parkingsystem/mvp/presenter/ReservationPresenterTest.java]: as you are testing the presenter you should not call the model, just call the methods on the presenter that set these values in the model

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/parkingsystem/mvp/model/ReservationModel.java (hunk 1 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/parkingsystem/mvp/model/ReservationModel.java]: fixed

File: app/src/test/java/com/example/parkingsystem/mvp/presenter/ReservationPresenterTest.java (hunk 2 above)
Comment (ezequielsetzesglobant) [reply app/src/test/java/com/example/parkingsystem/mvp/presenter/ReservationPresenterTest.java:42]: fixed

File: app/src/test/java/com/example/parkingsystem/mvp/presenter/ReservationPresenterTest.java (hunk 3 above)
Comment (ezequielsetzesglobant) [reply app/src/test/java/com/example/parkingsystem/mvp/presenter/ReservationPresenterTest.java]: fixed

**Review by BQuinteros — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request merged**